eng/checkout-job.yml

```diff
@@ -0,0 +1,60 @@
+### Check out job pulls down the GIT repository and packs its contents
+### into an Azure artifact for reuse by the subsequent build and test execution phases.
+jobs:
+- job: 'checkout_windows'
+  displayName: 'Checkout (Windows)'
+
+  variables:
+    use_open_queue: ${{ eq(variables['System.TeamProject'], 'public') }}
+
+  pool:
+    # Public Windows Build Pool
+    ${{ if variables.use_open_queue }}:
+      name: NetCorePublic-Pool
```

Comment:
Nit: We can most likely use the azure dev ops provided machines for the public jobs. Should put less strain on the build pool.

`windows-latest` should work.

Reply:
Fixed in 2nd commit.

Reply:
I think the syntax is actually `Hosted VS2017`

Reply:
Let's take this offline to our sync-up 3 PM meeting, it probably doesn't make sense for me to try out five more variants before I manage to hit the one you have in mind.